### PR TITLE
Implement pg_search

### PR DESCRIPTION
**Pull request by kljuni**
**1. Added pg_search Gem:**

- Integrated the pg_search gem to enhance search functionality within the application.

**2. Search Implementation:**

- Implemented tsearch and trigram (with two thresholds) search for improved search accuracy.

- Potential for further optimization by consolidating to a single threshold for trigram search in future iterations.

**3. Removed Algolia:**

- Deprecated the use of Algolia for search functionalities in favor of pg_search.

### CODE REVIEW

**Review by nmondollot — COMMENTED** [review]
Wow, I'm very impressed!
Excellent job! 🤩 

It's almost shippable as is, but I still have a few nitpicks:
- we can make the code a bit simpler (and easier to maintain) if we stick more to the "Stimulus way" using targets and actions (cf my inline comments)
  - we could even go deeper and use a turbo frame with HTML instead of JSON (since we don't need a search api for the foreseeable future), but honestly I'm fine with json too
- the low threshold seems a bit too low for area search
  - eg. a query for a problem like `Duel` returns areas like `Potala`, `Restant du Long Rocher` and `Rocher du Général` => not very helpful
  - but I guess we can tweak the values later
- Can we disable autocorrect on the search textbox?

A bit more advanced:
- When the server is slow to answer, if I do 2 searches consecutively I don't know if the results being displayed are for my first search or the second => it's a bit confusing.
  - how does algolia handle this? do they "cancel" previous searches?
- It'd be nice to have better accessibility for desktop: up/down arrows, enter to select, escape to close the modal, etc

**Comment by nmondollot** [comment]
Just had a look, it works really well 🤩 
Let's ship it when you're back!

Thank you for such great work 🙏

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,8 @@ gem 'sidekiq', '~> 7.1.2'
 gem 'redis', '~> 4.2.5'
 gem "rorvswild", "~> 1.6.4"
 gem 'bugsnag', '~> 6.25.2'
-gem 'algoliasearch-rails', '~> 2.2.2'
+gem 'pg_search', '~> 2.3.6'
+gem 'concurrent-ruby', '~> 1.2.3'
```
Comment (nmondollot) [inline Gemfile:19]: Why do you need the `concurrent-ruby` gem?

**Comment by kljuni** [comment]
Hi Nicolas,

Thanks for the feedback! I am back from vacation. Today I added a small fix
for managing focus when the mouse is already over some item when we input a
new query. I also removed the redundant concurrent-ruby (I used it in the
past for a possible solution).

All the best,
Ivan

**Comment by nmondollot** [comment]
It's live on prod!
It seems to work pretty well 👍 
I'll make more tests in the upcoming days.

Thank you so much for your contribution ❤️

**Pull request merged**